### PR TITLE
Implements doc search

### PR DESCRIPTION
**Pull request by arcanis**
**What's the problem this PR addresses?**

It's difficult at the moment to navigate the documentation since you need to know where to go.

**How did you fix it?**

Added a documentation based on Algolia.

@Haroenv I made two inputs, based on whether the display is mobile or desktop, but it seems the docsearch only matches one - is it expected?

### CODE REVIEW

File: packages/gatsby/gatsby-ssr.js [hunk 1]
```diff
@@ -1,7 +1,13 @@
-/**
- * Implement Gatsby's SSR (Server Side Rendering) APIs in this file.
- *
- * See: https://www.gatsbyjs.org/docs/ssr-apis/
- */
+import React from "react";
 
-// You can delete this file if you're not using it
+export function onRenderBody({setHeadComponents, setPreBodyComponents, setPostBodyComponents}) {
+  console.log(`foo`);
```
Comment (Haroenv) [inline packages/gatsby/gatsby-ssr.js]: Test thingie

File: packages/gatsby/gatsby-config.js [hunk 2]
```diff
@@ -34,6 +34,14 @@ module.exports = {
   plugins: [
     `gatsby-plugin-emotion`,
     `gatsby-plugin-react-helmet`,
+    {
+      resolve: `gatsby-plugin-algolia-docsearch`,
+      options: {
+        apiKey: `029f65f2c00301615fd14958b67d6730`,
+        indexName: `yarnpkg_next`,
+        inputSelector: `.docsearch`,
```
Comment (Haroenv) [inline packages/gatsby/gatsby-config.js]: You can give two different classes maybe, it should accept two selectors I think

File: packages/gatsby/gatsby-ssr.js (hunk 1 above)
Comment (Haroenv) [reply packages/gatsby/gatsby-ssr.js]:
```suggestion
```

File: packages/gatsby/gatsby-config.js (hunk 2 above)
Comment (arcanis) [reply packages/gatsby/gatsby-config.js]: From the look of it the script makes a `this.input = DocSearch.getInputFromSelector(inputSelector)`, so only one input 🤔

**Comment by arcanis** [comment]
Made an update to use a single input, but tbh it's not very pretty ... I'll probably revert

**Comment by Haroenv** [comment]
I was wrong, you can only have two dropdowns if you call `docsearch` twice. You might need to fork that Gatsby plugin or move the code into the library @arcanis

**Pull request merged**